### PR TITLE
Added internal written solution for cses 2185 Prime Multiples

### PR DESCRIPTION
**Pull request by alexyd88**
_If any of the below don't apply to this Pull Request, mark the checkbox as done._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions).
  - I understand that if it is clear that I have not attempted to follow these conventions, my PR will be closed.
  - If changes are requested, I will re-request a review after addressing them.

### CODE REVIEW

**Comment by alexyd88** [comment]
pre-commit.ci autofix

File: solutions/silver/cses-2185.mdx [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+source: CSES
+title: Prime Multiples
+author: Alex Du
+---
+
+## Solution
+
+We can identify that given any integer $c$, the number of integers that are divisible by it between 1 and $N$ is $\lfloor\frac{N}{c}\rfloor$
+
+If we consider each prime individually and find the number of multiples it has, we will find that there is an overlap
```
Comment (SansPapyrus683) [inline solutions/silver/cses-2185.mdx:12]: break this giant paragraph up

File: solutions/silver/cses-2185.mdx [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+Considering only $30$, the full example would be that our answer is added $3$ times for $2$, $3$, and $5$ (subsets of array with $1$ prime), then subtracted $3$
+times for $6$, $10$, and $15$ (subsets of array with $2$ primes), and finally added once for $30$ (subset of array with $3$ primes),
+overall only being counted once.
+
+To implement this, we would need to go through each subset of the array, and add the number of multiples it has if the number of primes is not divisible by 2, and
+subtract otherwise. This is known as the [Inclusion Exclusion Principle](https://cp-algorithms.com/combinatorics/inclusion-exclusion.html).
+In order to iterate through all subsets of the array, we can iterate from $1$ to $2^K-1$ and use bit operators to find which indices we are using.
+
+## Implementation
+Here is the code implementation.
```
Comment (SansPapyrus683) [inline solutions/silver/cses-2185.mdx:28]:
```suggestion
```
also add time complexity

File: solutions/silver/cses-2185.mdx [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+typedef long long ll;
+
+int main() {
+	ll N, K;
+	cin >> N >> K;
+	ll a[K];
+	for (auto &i : a) cin >> i;
```
Comment (SansPapyrus683) [inline solutions/silver/cses-2185.mdx:43]: plz don't use auto

File: solutions/silver/cses-2185.mdx [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+typedef long long ll;
+
+int main() {
+	ll N, K;
+	cin >> N >> K;
+	ll a[K];
```
Comment (SansPapyrus683) [inline solutions/silver/cses-2185.mdx:42]: just use a vector

File: solutions/silver/cses-2185.mdx [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+
+<CPPSection>
+
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+typedef long long ll;
+
+int main() {
+	ll N, K;
```
Comment (SansPapyrus683) [inline solutions/silver/cses-2185.mdx:40]: make these lowercase

File: solutions/silver/cses-2185.mdx [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+	cin >> N >> K;
+	ll a[K];
+	for (auto &i : a) cin >> i;
+	ll ans = 0;
+	for (ll i = 1; i < (1 << K); i++) {
+		ll primeProduct = 1;
```
Comment (SansPapyrus683) [inline solutions/silver/cses-2185.mdx:46]: snake_case

File: solutions/silver/cses-2185.mdx [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+		}
+		//__builtin_popcount gives the number of 1's in binary representation,
+		// which is also the number of primes we have multiplied
+		if (__builtin_popcount(i) % 2) {
+			ans += N / primeProduct;
+		} else {
+			ans -= N / primeProduct;
+		}
+	}
+	cout << ans << endl;
```
Comment (SansPapyrus683) [inline solutions/silver/cses-2185.mdx:67]: separate relevant segments w/ whitespace

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

File: solutions/silver/cses-2185.mdx [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+
+<CPPSection>
+
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+typedef long long ll;
+const int MAX_K = 20;
+
+ll a[MAX_K];
```
Comment (SansPapyrus683) [inline solutions/silver/cses-2185.mdx:43]: thought i asked for a vector

**Review by SansPapyrus683 — APPROVED** [review]

File: solutions/silver/cses-2185.mdx [hunk 9]
```diff
@@ -0,0 +1,77 @@
...
+				// ans doesn't change
+				if (prime_product > n / a[j]) {
+					prime_product = n + 1;
+					break;
+				}
+				prime_product *= a[j];
+			}
+		}
+
+		//__builtin_popcount gives the number of 1's in binary representation,
```
Comment (ryanchou-dev) [inline solutions/silver/cses-2185.mdx:63]: multiline comment this?

**Review by ryanchou-dev — APPROVED** [review]
yessir

File: solutions/silver/cses-2185.mdx (hunk 9 above)
Comment (SansPapyrus683) [reply solutions/silver/cses-2185.mdx:63]: two lines is fine i think

**Pull request merged**